jdaviz/utils.py

```diff
@@ -28,7 +28,23 @@ def __init__(self):
         # to give time for the app to load.
         self.first = True
 
-    def put(self, state, msg):
+    def put(self, state, msg, history=True, popup=True):
+        if msg.color not in ['info', 'warning', 'error', 'success', None]:
+            raise ValueError(f"color ({msg.color}) must be on of: info, warning, error, success")
+
+        if not msg.loading and history:
+            now = time.localtime()
+            timestamp = f'{now.tm_hour}:{now.tm_min:02d}:{now.tm_sec:02d}'
```

Comment:
```python
from datetime import datetime
```
```suggestion
            timestamp = datetime.now().strftime('%H:%M:%S')
```

Reply:
I just did this since time is already imported.  Are there significant advantages worth the extra import?

Reply:
`datetime` is already standard lib and in case you want different formatting in the future or something, no need to reinvent the wheel. But if you want to keep this, that is fine too. Just a suggestion.